### ISSUE TITLE
Should init script purge EASYBUILD_* environment variables?

### ISSUE THREAD

**Issue by casparvl** [EESSI/software-layer#58]
**Issue**

I ran into this error when running from the EESSI container:

```
Singularity> source /cvmfs/pilot.eessi-hpc.org/2020.12/init/bash
[EESSI pilot 2020.12] $ module load EasyBuild
[EESSI pilot 2020.12] $ eb -S ReFrame
ERROR: Failed to parse configuration options: 'parseconfigfiles: configfile /sw/noarch/RedHatEnterpriseServer7/2020//software/eb/4.3.2/etc/config.cfg not found.'
```
Whoops, that's because our host system has `EASYBUILD_CONFIGFILES` set up.

How should we go about this? Should the init script unset any `EASYBUILD_*` variables?

**Comment by boegel** [comment]
This is a tough one, and my gut feeling says no.

You could ask the same question about other environment variables, like `$MODULEPATH`, `$PATH`, `$PYTHONPATH`, etc.

The purpose of the EESSI init script is not to drop you in a clean environment, but just make the necessary changes to make the software provided by EESSI accessible.

**Comment by casparvl** [comment]
I agree with that, it shouldn't be the init script. In fact, if you were running this stuff natively, it wouldn't even lead to problems: I'd get the EasyBuild installation from `EESSI`, but with our local configuration. That's kind of expected.

Thinking about it more, I guess the thing that felt unintuitive is the fact that the behaviour of the _container_ is affected by these environment variables. In my mind, containers _should_ ideally run the same, where-ever you deploy them. Also, your same argument doesn't (fully) apply there: the container has it's own `$PATH`, `$LD_LIBRARY_PATH` etc. Other variables like `MODULEPATH` and `PYTHONPATH` are inherited from the host.

Then again, maybe I just have unrealistic expectations of containers =) I don't know, maybe someone with more experience with containers should answer the question whether it is possible (or sensible) for a container to clean the host's environment variables upon container startup.

**Comment by boegel** [comment]
You can use `--cleanenv` with `singularity shell` to avoid environment variables leaking into the container, just like you can use an alternate home directory using something like `--home /tmp/$USER/home_for_container`

**Comment by ocaisa** [comment]
`EESSI-extend` would like that you _do_ set some settings (like where to store sources), so I agree that we in general don't want purging. If there are specific settings that are problematic, they can be added to `EESSI-extend` when problems arise because of them.
